src/huggingface_hub/_login.py

```diff
@@ -89,7 +89,7 @@ def login(token: Optional[str] = None, add_to_git_credential: bool = False) -> N
     elif is_notebook():
         notebook_login()
     else:
-        interpreter_login()
+        interpreter_login(token)
```

Comment:
I would rather prefer to put the logic inside `interpreter_login` directly so that the logic stays the same no matter if it's called from the CLI or from a script.

```suggestion
        interpreter_login()
```